src/utils.ts

```diff
@@ -219,6 +261,139 @@ export interface Tx {
+        txid: tx.txHash,
+      };
+    });
+
+  const attestations: { [key: string]: AttestationReceipt<"WormholeCore"> } =
+    {};
+  for (const vaaType of possibleVaaTypes) {
+    const key = `${vaaType}SignedVaa`;
+    if (key in txStatus && txStatus[key as keyof TransactionStatus] !== null) {
+      const vaa = deserialize(
+        "Uint8Array",
+        encoding.hex.decode(txStatus[key as keyof TransactionStatus])
+      );
+
+      attestations[vaaType] = {
+        id: {
+          emitter: vaa.emitterAddress,
+          sequence: vaa.sequence,
+          chain: vaa.emitterChain,
+        },
+        attestation: vaa,
+      };
+    }
+  }
+
+  switch (state) {
+    case TransferState.SourceInitiated:
+      // Initital transfer vaa from source chain
+      if ("transfer" in attestations && attestations["transfer"]) {
+        return {
+          from: srcChain,
+          to: dstChain,
+          originTxs,
+          state: TransferState.SourceFinalized,
+          attestation: attestations["transfer"],
+        } satisfies SourceFinalizedTransferReceipt<
+          AttestationReceipt<"WormholeCore">
+        >;
+      }
+      break;
+
+    case TransferState.DestinationInitiated:
+      // VAA to be redeemed on dest chain
+      if ("redeem" in attestations && attestations["redeem"]) {
+        return {
+          from: srcChain,
+          to: dstChain,
+          originTxs,
+          destinationTxs,
+          state,
+          attestation: attestations["redeem"],
+        } satisfies CompletedTransferReceipt<
+          AttestationReceipt<"WormholeCore">
+        >;
+      }
+
+      // Initial transfer vaa from orgin chain
+      if ("transfer" in attestations && attestations["transfer"]) {
+        return {
+          from: srcChain,
+          to: dstChain,
+          originTxs,
+          destinationTxs,
+          state,
+          attestation: attestations["transfer"],
+        } satisfies CompletedTransferReceipt<
+          AttestationReceipt<"WormholeCore">
+        >;
+      }
+
+      break;
+
+    case TransferState.Failed:
+      // VAA that is used to refund on source chain
+      if ("refund" in attestations && attestations["refund"]) {
+        return {
+          from: srcChain,
+          to: dstChain,
+          originTxs,
+          destinationTxs,
+          state: TransferState.Failed,
+          attestation: attestations["refund"],
+          error: "Refunded on source chain",
+        } satisfies FailedTransferReceipt<AttestationReceipt<"WormholeCore">>;
+      }
+
+      // No vaa to refund on source chain
+      return {
+        from: srcChain,
+        to: dstChain,
+        originTxs,
+        destinationTxs,
+        state: TransferState.Failed,
+        error: "Failed to complete transfer",
+      } satisfies FailedTransferReceipt<AttestationReceipt<"WormholeCore">>;
+  }
+
+  return {
+    from: srcChain,
+    to: dstChain,
+    originTxs,
+    state: TransferState.SourceInitiated,
```

Comment:
do we expect this to ever hit? if not, should we add a `console.warn`? should we change this to just return the existing value of `state`?

Reply:
yes, this will probably be the most common one depending on how frequently its polled. Hit before any VAA is signed or it fails on a timeout